firedrake/preconditioners/massinv.py

```diff
@@ -0,0 +1,80 @@
+from firedrake.preconditioners.base import PCBase
+from firedrake.petsc import PETSc
+
+__all__ = ("MassInvPC", )
+
+
+class MassInvPC(PCBase):
```

Comment:
It would be good if this used SLATE in the DG case to directly calculate the inverse mass matrix. (I will do this some time, if no one else does.)